vividus-plugin-browserstack/src/main/java/org/vividus/browserstack/BrowserStackAppAutomateClient.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2019-2019-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.browserstack;
+
+import java.net.URI;
+import java.time.Duration;
+import java.util.List;
+
+import com.browserstack.appautomate.AppAutomateClient;
+import com.browserstack.client.BrowserStackRequest;
+import com.browserstack.client.exception.BrowserStackException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.vividus.util.UriUtils;
+import org.vividus.util.json.JsonUtils;
+import org.vividus.util.wait.DurationBasedWaiter;
+import org.vividus.util.wait.WaitMode;
+import org.vividus.util.wait.Waiter;
+
+public class BrowserStackAppAutomateClient extends AppAutomateClient
```

Comment:
do you need inheritance? can't you use composition here?

Reply:
This client will be used later for getting logs for running test suite